lib/rubocop/cop/performance/regexp_match.rb

```diff
@@ -154,6 +154,9 @@ def message(node)
         def last_match_used?(match_node)
           scope_root = scope_root(match_node)
           body = scope_root ? scope_body(scope_root) : match_node.ancestors.last
+
+          return true if modifier_if_or_unless?(body.source, match_node.source)
```

Comment:
I guess the implementation has a false positive. The cop will ignore the following code.

```ruby
def foo
  return if /re/ =~ str
  do_something Regexp.last_match[1]
end
```

Reply:
I'm not sure I understand this problem.
Is it this problem even at the time of RuboCop 0.52.1?
I don't understand the expected behavior and the actual behavior 💦